pkg/storage/etcd/testing/utils.go

```diff
@@ -164,8 +311,8 @@ func (m *EtcdTestServer) Terminate(t *testing.T) {
 }
 
 // NewEtcdTestClientServer creates a new client and server for testing
-func NewEtcdTestClientServer(t *testing.T) *EtcdTestServer {
-	server := configureTestCluster(t, "foo")
+func NewEtcdTestClientServer(t *testing.T, secure bool) *EtcdTestServer {
+	server := configureTestCluster(t, "foo", secure)
```

Comment:
@timothysc that would be great if we enable secure by default. The tests I add are not that good, I just want to find a better way to do the test. Thanks :)